.github/workflows/dist.yml

```diff
@@ -0,0 +1,48 @@
+# vim:ts=2:sw=2:et:ai:sts=2
+name: 'Build distribution'
+
+on:
+  # Only run when release is created in the master branch
+  release:
+    types: [created]
+    branches:
+      - 'master'
+
+jobs:
+  build_sdist:
+    name: 'Build source distribution'
+    runs-on: 'ubuntu-latest'
+    steps:
+      - uses: actions/checkout@v3
+        name: 'Checkout source repository'
+        with:
+            # You need to include this or setuptools_scm in GitHub runners won't detect the version
+            fetch-depth: 0
+
+      - uses: actions/setup-python@v4
+
+      - name: 'Build sdist'
+        run: python3 setup.py sdist
+
+      - uses: actions/upload-artifact@v2
+        name: 'Upload build artifacts'
+        with:
+          path: 'dist/*.tar.gz'
+
+  upload_pypi:
```

Comment:
This would need to use an environment for secrets. It's usually `release`.

Reply:
Added `environment: release`.